rye/src/pyproject.rs

```diff
@@ -1107,6 +1119,14 @@ fn is_rye_managed(doc: &Document) -> bool {
         .unwrap_or(false)
 }
 
+fn is_workspace_only(doc: &Document) -> bool {
+    doc.get("tool")
+        .and_then(|x| x.get("rye"))
+        .and_then(|x| x.get("workspace-only"))
+        .and_then(|x| x.as_bool())
+        .unwrap_or(false)
```

Comment:
would this config variable live better inside the existing group called `tool.rye.workspace`?

Reply:
What about instead of adding a workspace only value we just expect members = [] where members is empty? So it's implied to be used as a workspace without members to install.

Reply:
I guess there are different ways to go but I was thinking that a workspace-only can be used both with members and without. And it would always be a workspace, hence the workspace key should be set anyway.

Reply:
The `tool.rye.workspace` configuration is not normally needed for single project development. The `workspace-only` here might be more appropriately called `no-package`? or `is-package = true` ?

Reply:
To just keep the discussion diverse, I kinda think about this as everything is a workspace. Some workspaces you want to add packages/projects (members) to. Some of those projects/packages (members) are meant to be installed or their environments are managed via a workspace .venv.

So 
> - python package

What keeps us from considering it as

> - python package and a workspace

and

> - workspace (only)

Is just a workspace with no added members.

Reply:
I appreciate that. A workspace is a ceiling, or the top level of a thing. Can't include workspaces in workspaces. Then it would seem anything that's is "the top" and is never installed into something else, is a workspace.